tests/http/test_proxyagent.py

```diff
@@ -618,8 +616,8 @@ def _do_https_request_via_proxy(
         request.finish()
 
         # now we make another test server to act as the upstream HTTP server.
-        server_ssl_protocol = _wrap_server_factory_for_tls(
-            _get_test_protocol_factory()
+        server_ssl_protocol = wrap_server_factory_for_tls(
+            _get_test_protocol_factory(), self.reactor, sanlist=[b"DNS:test.com"]
```

Comment:
Is passing in an explicit sanlist here a meaningful change?

Reply:
Ditto below in the rest of this test

Reply:
No: `_wrap_server_factory_for_tls` defaulted to this if no `sanlist` argument was provided.